Source/WebKit/UIProcess/Extensions/Cocoa/WebExtensionContextCocoa.mm

```diff
@@ -299,52 +235,38 @@ static WKWebExtensionContextError toAPI(WebExtensionContext::Error error)
 
 void WebExtensionContext::clearError(Error error)
 {
-    if (!m_errors.get().count)
-        return;
-
-    auto errorCode = toAPI(error);
-    auto *indexes = [m_errors indexesOfObjectsPassingTest:^BOOL(NSError *error, NSUInteger, BOOL *) {
-        return error.code == errorCode;
-    }];
-
-    if (!indexes.count)
+    if (!m_errors.size())
         return;
 
+    auto errorCode = toAPIError(error);
     [wrapper() willChangeValueForKey:@"errors"];
-    [m_errors removeObjectsAtIndexes:indexes];
+    m_errors.removeAllMatching([&](auto& error) {
+        if (error->errorCode() == errorCode)
+            return true;
+        return false;
+    });
     [wrapper() didChangeValueForKey:@"errors"];
 
     dispatch_async(dispatch_get_main_queue(), makeBlockPtr([this, protectedThis = Ref { *this }]() {
         [NSNotificationCenter.defaultCenter postNotificationName:WKWebExtensionContextErrorsDidUpdateNotification object:wrapper() userInfo:nil];
     }).get());
 }
 
-NSArray *WebExtensionContext::errors()
+bool WebExtensionContext::load(WebExtensionController& controller, String storageDirectory, RefPtr<API::Error>& outError)
```

Comment:
This applies everywhere, but we should switch to `Expected<bool, RefPtr<API::Error>>` and drop `outError`. This can be a follow-up, unless you find it easier to do now. (Same would apply to `WebExtension` class.)